src/components/CharacterList.js

```diff
@@ -1,14 +1,23 @@
-import React from "react";
+import React from 'react';
 
-import Character from "./Character";
+import Character from './Character';
 
 const CharacterList = props => {
+  // I spent a very long time tryng to debug this project. It turns out it was set up wrong in the first place. In addition to a change in reducers/index.js, it needed 'props.characters.characters.map' on line 11:
   return (
-    <ul>
-      {props.characters.map(character => {
-        return <Character key={character.name} character={character} />;
-      })}
-    </ul>
+    <div>
+      {props.characters && (
+        <ul>
+          {props.characters.characters.map(character => {
+            return <Character key={character.name} character={character} />;
+          })}
+        </ul>
+      )}
+      {props.error && <p className='error'>{props.error}</p>}
+      {!props.characters && !props.error && (
+        <p>Component not rerendering upon FETCH_SWCHARS_SUCCESS</p>
```

Comment:
technically yes, but since this is also what will display when you don't have the characters and there is no error yet, this will be the display message for "Loading, please wait..." or something along those lines, even if it displays only for a split second